hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/hudi/HoodieIncrementalFileIndex.scala

```diff
@@ -36,9 +36,10 @@ class HoodieIncrementalFileIndex(override val spark: SparkSession,
                                  override val schemaSpec: Option[StructType],
                                  override val options: Map[String, String],
                                  @transient override val fileStatusCache: FileStatusCache = NoopCache,
-                                 override val includeLogFiles: Boolean)
+                                 override val includeLogFiles: Boolean,
+                                 override val shouldEmbedFileSlices: Boolean)
```

Comment:
do we have some bootstrap tests covering the path w/ and w/o `shouldEmbedFileSlices`?